source/LifecycleMethod.cpp

```diff
@@ -319,26 +386,31 @@ const DexMethod* MT_NULLABLE LifecycleMethod::create_dex_method(
   mt_assert(dex_klass != nullptr);
 
   int callee_count = 0;
-  for (const auto& callee : callees_) {
-    auto* dex_method = callee.get_dex_method(dex_klass);
-    if (!dex_method) {
-      // Dex method does not apply for current APK.
-      // See `LifecycleMethod::validate()`.
-      continue;
-    }
 
-    ++callee_count;
+  if (const auto* callees = std::get_if<std::vector<LifecycleMethodCall>>(&body_)) {
+    for (const auto& callee : *callees) {
+      auto* dex_method = callee.get_dex_method(dex_klass);
+      if (!dex_method) {
+        // Dex method does not apply for current APK.
+        // See `LifecycleMethod::validate()`.
+        continue;
+      }
+
+      ++callee_count;
 
-    std::vector<Location> invoke_with_registers{this_location};
-    auto* type_list = callee.get_argument_types();
-    // This should have been verified at the start of `create_methods`
-    mt_assert(type_list != nullptr);
-    for (auto* type : *type_list) {
-      auto argument_register = method.get_local(type_index_map.at(type));
-      invoke_with_registers.push_back(argument_register);
+      std::vector<Location> invoke_with_registers{this_location};
+      auto* type_list = callee.get_argument_types();
+      // This should have been verified at the start of `create_methods`
+      mt_assert(type_list != nullptr);
+      for (auto* type : *type_list) {
+        auto argument_register = method.get_local(type_index_map.at(type));
+        invoke_with_registers.push_back(argument_register);
+      }
+      main_block->invoke(
+          IROpcode::OPCODE_INVOKE_VIRTUAL, dex_method, invoke_with_registers);
     }
-    main_block->invoke(
-        IROpcode::OPCODE_INVOKE_VIRTUAL, dex_method, invoke_with_registers);
```

Comment:
maybe move these in a function `generate_invoke_instructions` that take a list of `LifecycleMethodCall` and the `main_block`. This will be useful for the next step.